lib/phoenix_html/tag.ex

```diff
@@ -99,22 +100,31 @@ defmodule Phoenix.HTML.Tag do
       attr_name = "#{attr}-#{dasherize(k)}"
       case is_list(v) do
         true  -> nested_attrs(attr_name, v, acc)
-        false -> [{attr_name, v}|acc]
+        false -> Map.put(acc, attr_name, v)
       end
     end
   end
 
   defp build_attrs(_tag, []), do: []
-  defp build_attrs(tag, attrs), do: build_attrs(tag, attrs, [])
-
-  defp build_attrs(_tag, [], acc),
-    do: acc |> Enum.sort |> tag_attrs
+  defp build_attrs(tag, attrs), do: build_attrs(tag, attrs, %{})
+
+  defp build_attrs(_tag, [], acc) do
+    Enum.map(acc, fn
+      {k, v} when is_atom(k) -> {Atom.to_string(k), v}
+      {k, v} when is_bitstring(k) -> {k, v}
+    end)
```

Comment:
If anything, we should test for `is_binary`, arbitrary bitstrings would probably crash somewhere down the road